packages/api/exec.mts

```diff
@@ -135,38 +157,34 @@ export function tsx(options: NodeRequestType) {
  * Install a specific package:
  *
  *     npmInstall({
- *       cwd: '/Users/ben/.srcbook/foo',
- *       package: 'marked',
+ *       cwd: '/path/to/project',
+ *       packages: ['lodash'],
  *       stdout(data) {console.log(data.toString('utf8'))},
  *       stderr(data) {console.error(data.toString('utf8'))},
  *       onExit(code) {console.log(`Exit code: ${code}`)}
  *     });
- *
  */
-export function npmInstall(options: NPMInstallRequestType) {
-  const { cwd, stdout, stderr, onExit } = options;
-  const args = options.packages
-    ? ['install', '--include=dev', ...(options.args || []), ...options.packages]
-    : ['install', '--include=dev', ...(options.args || [])];
-
-  return spawnCall({
+export const npmInstall = ({ cwd, packages = [], args = [], ...rest }: NPMInstallRequestType) =>
+  spawnCall({
     command: 'npm',
     cwd,
-    args,
-    stdout,
-    stderr,
-    onExit,
+    args: ['install', '--include=dev', ...args, ...packages],
     env: process.env,
+    ...rest,
   });
-}
 
 /**
  * Run vite.
  */
-export function vite(options: NpxRequestType) {
-  return spawnCall({
-    ...options,
-    command: Path.join(options.cwd, 'node_modules', '.bin', 'vite'),
-    env: process.env,
+export const vite = ({ cwd, args = [], env = process.env, ...rest }: NpxRequestType) =>
+  spawnCall({
+    command:
+      process.platform === 'win32' ? 'npx.cmd' : Path.join(cwd, 'node_modules', '.bin', 'vite'),
+    cwd,
+    args: process.platform === 'win32' ? ['vite', ...args] : args,
+    env: {
+      ...env,
+      FORCE_COLOR: '1',
```

Comment:
What's this?

Reply:
..env is for copying all the object, for creating in the following object although I added FORCE_COLOR: '1' Because it was getting difficult to find the response so for making it shine I added, I will remove it if you want